simtools/model/model_parameter.py

```diff
@@ -0,0 +1,561 @@
+        ----------
+        par_name: str
+            Name of the parameter.
+        value:
+            Value of the parameter.
+        is_file: bool
+            Indicates whether the new parameter is a file or not.
+        is_applicable: bool
+            Indicates whether the new parameter is applicable or not.
+
+        Raises
+        ------
+        InvalidModelParameter
+            If an existing parameter is tried to be added.
+        """
+        if par_name in self._parameters:
+            msg = f"Parameter {par_name} already in the model, use change_parameter instead"
+            self._logger.error(msg)
+            raise InvalidModelParameter(msg)
+
+        self._logger.info(f"Adding {par_name}={value} to the model")
+        self._parameters[par_name] = {}
+        self._parameters[par_name]["value"] = value
+        self._parameters[par_name]["type"] = type(value)
+        self._parameters[par_name]["applicable"] = is_applicable
+        self._parameters[par_name]["file"] = is_file
+
+        self._is_config_file_up_to_date = False
+        if is_file:
+            self._is_exported_model_files_up_to_date = False
+
+    def change_parameter(self, par_name, value):
+        """
+        Change the value of an existing parameter to the model. This function does not modify the \
+        DB, it affects only the current instance.
+
+        Parameters
+        ----------
+        par_name: str
+            Name of the parameter.
+        value:
+            Value of the parameter.
+
+        Raises
+        ------
+        InvalidModelParameter
+            If the parameter to be changed does not exist in this model.
+        """
+        if par_name not in self._parameters:
+            msg = f"Parameter {par_name} not in the model, use add_parameters instead"
+            self._logger.error(msg)
+            raise InvalidModelParameter(msg)
+
+        type_of_par_name = locate(self.get_parameter_type(par_name))
+        if not isinstance(value, type_of_par_name):
+            self._logger.warning(
+                f"The type of the provided value ({value}, {type(value)}) "
+                f"is different from the type of {par_name} "
+                f"({self.get_parameter_type(par_name)}). "
+                f"Attempting to cast to the correct type."
+            )
+            try:
+                value = type_of_par_name(value)
+            except ValueError:
+                self._logger.error(
+                    f"Could not cast {value} to {self.get_parameter_type(par_name)}."
+                )
+                raise
+
+        self._logger.debug(
+            f"Changing parameter {par_name} "
+            f"from {self.get_parameter_value(par_name)} to {value}"
+        )
+        self._parameters[par_name]["value"] = value
+
+        # In case parameter is a file, the model files will be outdated
+        if self.get_parameter_file_flag(par_name):
+            self._is_exported_model_files_up_to_date = False
```

Comment:
OK. But in most cases we use the model as a whole and not individual parameters. So one change makes a model being outdated. But let's discuss your proposal again and see if we should update this accordingly in a future PR.

Reply:
Yes. Let's leave this comment open for a while, I will have a look if I find a quick solution on Monday. Once simtools is close to complete and we have all the necessary derivation functions for all parameters, we would like to know which ones are not updated and update only those, because the derivation of some parameters might take some time (e.g. the random angle tool).

Reply:
Ok, as we spoke in person, this is not necessary, as the workflows will automatically update all parameters related to a parameter changed.